contracts/protocols/polygon/uniswapStaking/main.sol

```diff
@@ -120,52 +122,67 @@ contract Resolver is Helpers {
+            rewardDetails[i].token0 = pool.token0();
+            rewardDetails[i].token1 = pool.token1();
 
-        (rewardDetails.maxAmount0, rewardDetails.maxAmount1) = LiquidityAmounts.getAmountsForLiquidity(
-            sqrtRatioX96,
-            TickMath.getSqrtRatioAtTick((int24(maxLowerTick))),
-            TickMath.getSqrtRatioAtTick((int24(maxUpperTick))),
-            uint128(rewardDetails.liquidity)
-        );
+            uint160 sqrtRatioX96;
+            (sqrtRatioX96, rewardDetails[i].tick, , , , , ) = pool.slot0();
+            rewardDetails[i].liquidity = pool.liquidity();
+            (rewardDetails[i].minAmount0, rewardDetails[i].minAmount1) = LiquidityAmounts.getAmountsForLiquidity(
+                sqrtRatioX96,
+                TickMath.getSqrtRatioAtTick((int24(tickInfo[i].minLowerTick))),
+                TickMath.getSqrtRatioAtTick((int24(tickInfo[i].minUpperTick))),
+                uint128(rewardDetails[i].liquidity)
+            );
 
-        if (key.startTime < block.timestamp) {
-            bytes32 incentiveId = keccak256(abi.encode(key));
+            (rewardDetails[i].maxAmount0, rewardDetails[i].maxAmount1) = LiquidityAmounts.getAmountsForLiquidity(
+                sqrtRatioX96,
+                TickMath.getSqrtRatioAtTick((int24(tickInfo[i].maxLowerTick))),
+                TickMath.getSqrtRatioAtTick((int24(tickInfo[i].maxUpperTick))),
+                uint128(rewardDetails[i].liquidity)
+            );
 
-            Incentive memory incentive;
-            (incentive.totalRewardUnclaimed, incentive.totalSecondsClaimedX128, incentive.numberOfStakes) = staker
-                .incentives(incentiveId);
+            if (tickInfo[i].key.startTime < block.timestamp) {
+                bytes32 incentiveId = keccak256(abi.encode(tickInfo[i].key));
 
-            uint256 totalSecondsUnclaimedX128 = ((Math.max(key.endTime, block.timestamp) - key.startTime) << 128) -
-                incentive.totalSecondsClaimedX128;
-            rewardDetails.rate = uint256(
-                (incentive.totalRewardUnclaimed * 0x100000000000000000000000000000000) /
-                    uint256(totalSecondsUnclaimedX128)
-            );
+                Incentive memory incentive;
+                (incentive.totalRewardUnclaimed, incentive.totalSecondsClaimedX128, incentive.numberOfStakes) = staker
+                    .incentives(incentiveId);
+
+                uint256 totalSecondsUnclaimedX128 = ((Math.max(tickInfo[i].key.endTime, block.timestamp) -
+                    tickInfo[i].key.startTime) << 128) - incentive.totalSecondsClaimedX128;
+                rewardDetails[i].rate = uint256(
+                    (incentive.totalRewardUnclaimed * 0x100000000000000000000000000000000) /
+                        uint256(totalSecondsUnclaimedX128)
+                );
+            }
+        }
+    }
+
+    function getTicks(address[] memory pools) public view returns (int24[] memory ticks, int24[] memory tickspacings) {
```

Comment:
suggestion: return tick and tickspacing in a struct.

Reply:
Sure.